src/context/videos-context.js

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState
+} from "react";
+import videoDB from "../database/videoDataBase";
+import videosReducer from "./videosReducer";
+
+const VideosContext = createContext();
+
+export function VideosContextProvider({ children }) {
+  const [state, dispatch] = useReducer(videosReducer, {
+    videos: [],
+    liked: [],
+    watchLater: [],
+    history: [],
+    playlist: [
+      {
+        name: "myplaylist",
+        videosAdded: []
+      }
+    ]
+  });
+
+  const [isLoading, setIsLoading] = useState("fetchStarted");

```

Comment:
you can give isLoading initialstate as true and false

Reply:
forgot remove this, as in this version have not done loading feature. will have this feature in the upcoming version so removed it